systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/RoleResource.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.resources.kubernetes;
+
+import io.fabric8.kubernetes.api.model.rbac.Role;
+import io.fabric8.kubernetes.api.model.rbac.RoleBuilder;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.resources.ResourceManager;
+import io.strimzi.systemtest.resources.ResourceType;
+import io.strimzi.test.TestUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class RoleResource implements ResourceType<Role> {
+
+    private static final Logger LOGGER = LogManager.getLogger(RoleResource.class);
+
+    @Override
+    public String getKind() {
+        return Constants.ROLE;
+    }
+    @Override
+    public Role get(String namespace, String name) {
+        return ResourceManager.kubeClient().namespace(namespace).getRole(name);
+    }
+    @Override
+    public void create(Role resource) {
+        ResourceManager.kubeClient().namespace(resource.getMetadata().getNamespace()).createOrReplaceRole(resource);
+    }
+    @Override
+    public void delete(Role resource) {
+        ResourceManager.kubeClient().namespace(resource.getMetadata().getNamespace()).deleteRole(resource.getMetadata().getName());
```

Comment:
I think better would be use deleteRole with signature `deleteRole(namespaceName, resourceName)` inside KubeClient. I know in my refactor that these types of methods without specifying namespace explicitly end up with race-condition...

Reply:
Done